packages/block-editor/src/components/list-view/use-clipboard-handler.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useRefEffect } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import { useNotifyCopy } from '../../utils/use-notify-copy';
+import { focusListItem } from './utils';
+import { getPasteBlocks, setClipboardBlocks } from '../writing-flow/utils';
+
+// This hook borrows from useClipboardHandler in ../writing-flow/use-clipboard-handler.js
+// and adds behaviour for the list view, while skipping partial selection. In the future,
+// consider refactoring the two hooks to share more code.
```

Comment:
You can drop the part about "in the future," if you want. It think this is probably as good as it will get 😀

Reply:
Ha! Good point 😄
I'll remove that line.